### ISSUE TITLE
.obj are not packaged

### ISSUE THREAD

**Issue by dmnsgn** [electron-userland/electron-builder#6126]
* **Electron-Builder Version**: 22.10.5

* **Node Version**: 16.x

* **Electron Version**: 13
* **Electron Type (current, beta, nightly)**: current

* **Target**: win / mac / linux

What's the rationale behind excluding `.obj` files?

https://github.com/electron-userland/electron-builder/blob/c623279a31d3d203ad1f150d83905a7e1f58a731/packages/app-builder-lib/src/util/appFileCopier.ts#L172

**Comment by dmnsgn** [comment]
Yes, still relevant. Awaiting any feedback on:

> What's the rationale behind excluding .obj files?

**Comment by dmnsgn** [comment]
It is still the case.

**Comment by dmnsgn** [comment]
bump

**Comment by dmnsgn** [comment]
Still valid

**Comment by mmaietta** [comment]
>What's the rationale behind excluding .obj files?

No clue on rationale for it, it's a 7yr old commit.

**Comment by throni3git** [comment]
i have the same issue. as i understand form the context of the code it is to exclude build artifacts from being put into an Electron app - but in my case it is the Wavefront OBJ-format for 3D meshes. somehow (opposed to this being a 7yr old commit) it worked in `electron-builder v25`, but with `v26` the files are not copied any more.

is this the issue you are also having, @dmnsgn ?

**Comment by dmnsgn** [comment]
Yes same reason, I also need to include Wavefront `.obj` in my app.

**Comment by mmaietta** [comment]
Wait a sec, I just remembered we have a some workarounds that were added for odd/edge cases like this.

Please try either:

Config prop `disabledefaultignoredfiles` (note, you'll need to fully define/copy the other ignored extensions and excluded files since this completely disables all)
https://www.electron.build/app-builder-lib.interface.configuration#disabledefaultignoredfiles
Should interface with this: 
https://github.com/electron-userland/electron-builder/blob/836a15c6c70abf8582aaa63603e14f77d5fa3f89/packages/app-builder-lib/src/fileMatcher.ts#L14-L19

If this `.obj` is in a node_module, then you can return true from this build hook to force include the files. Return false to use default logic
https://www.electron.build/app-builder-lib.interface.configuration#onnodemodulefile

**Comment by throni3git** [comment]
thanks for the reply! i tried `disableDefaultIgnoredFiles` but it did not help. here is my `electron-builder.ts`:

```typescript
import { Configuration } from "electron-builder";

const configElectronBuilder: Configuration = {
  appId: "com.company.myapp",
  productName: "MyApp",
  icon: "./public/logo/bounce/256x256.png",
  files: ["start-electron.js", "dist/**/*"],
  disableDefaultIgnoredFiles: true,
  win: {
    icon: "./public/logo/generated/myapp.ico",
  },
  asar: false,
  linux: {
    executableName: "MyApp",
    category: "Utility",
    target: ["dir", "AppImage"],
  },
  directories: {
    buildResources: "resources",
    output: "release",
  },
  onNodeModuleFile: () => true, // tried this as well
};

export default configElectronBuilder;
```

setting `disableDefaultIgnoredFiles` to `true` had no effect. i run `vite build && electron-builder --config electron-builder.ts`, so basically it takes the transpiled output from `dist` folder which contain the `.obj` files. in my situation it is not about `obj` files being in the `node_modules`.

is there anything else i can do? i hope that insights for my case help others too, like dmnsgn :)

btw, i updated from electron-builder v25 to v26.3 (had to update it manually to this minor version)

**Comment by mbotsch** [comment]
Any updates on this issue? It's the same for me, wanting to include 3D models in OBJ format.

**Comment by throni3git** [comment]
i think i am just going the hack-around way and rename my *.obj-files to *.objobjobj-files 🪿

**Comment by mmaietta** [comment]
Got a refactor incoming to resolve this for v27. Removes the non-extensible `disabledefaultignoredfiles` as well. Default exclusions will apply BUT can now be explicitly overridden by a `files` glob. `log.warn` is added to make sure nothing unexpected occurs (e.g. `.github` folder gets included because someone decides `**/.github` is a great idea in `files` 🤷  - best to be logging-forward and allow explicit control though)

Solution in PR https://github.com/electron-userland/electron-builder/pull/9954 if you want to TAL
